lib/concurrent/synchronization/java_pure_object.rb

```diff
@@ -32,6 +33,10 @@ def ns_signal
           JRuby.reference0(self).notify
           self
         end
+
+        def ensure_ivar_visibility!
+          # relying on undocumented behavior of JRuby, ivar access is volatile
```

Comment:
I'm specifically talking about the case where adding a new ivar to an object in JRuby which causes it to resize the object - that is I believe a volatile operation - I think we spent a while talking about that in the past.